### DISCUSSION TITLE
Open options icon and vertical folder toolbar.

### DISCUSSION THREAD

**Discussion by darkj2k** [YoshifumiFuyuno/Search-from-Popup-or-ContextMenu#60] [Q&A] [answered]
The icon of open options button don't has option for replace icon. Possible to changing it by CSS with base64 icon?
![options](https://user-images.githubusercontent.com/3615387/196038271-b404fe7e-fdc0-41fb-8fc1-d7df169aae38.jpg)

If the folder with lots of search engines inside, the horizontal toolbar with search engine name showing will be too long. Possible change folder toolbar from horizontal to vertical?
![horizontal to vertical](https://user-images.githubusercontent.com/3615387/196038491-a78f9092-d06d-4dea-8470-3cfbf17a5a6d.jpg)

Also possible to changing the background color of folder toolbar? I want to making it a little different with top-level toolbar background color when it popup. Thanks in advance.

**Comment by YoshifumiFuyuno** [comment, marked as answer]
* open Options
  Of course you can. Just change the "Shop" part of this to "Options".
  https://github.com/YoshifumiFuyuno/Search-from-Popup-or-ContextMenu/discussions/56#discussioncomment-3780987
  
  If you want to examine it yourself, look at the Inspector (press F12) and the "data-name=" part is the string you should specify.
  You can also edit the CSS and see the results in real time.
  https://firefox-source-docs.mozilla.org/devtools-user/page_inspector/
  https://developer.chrome.com/docs/devtools/overview/

* Change the display in folders from horizontal to vertical
  ```CSS
  .spc-eFolder.s-iconAndName {
    --col: 1;
  }
  ```
  However, if you add the following margin, try it for yourself, such as 1.1, and change it accordingly.
  https://github.com/YoshifumiFuyuno/Search-from-Popup-or-ContextMenu/discussions/56#discussioncomment-3878410

* Change the background inside the folder
  ```CSS
  .spc-eFolder {
    background-color: red;
  }
  ```
  or
  ```CSS
  .spc-eFolder {
    background: linear-gradient(45deg,rgba(150,251,196,0.8) 0%,rgba(249,245,134,0.8) 100%);
  }
  ```

**Reply by darkj2k** [reply]
Change the display in folders from horizontal to vertical is not working.

```
.spc-eFolder.s-iconAndName {
  --col: 1;
}
```
It is still display as horizontal.
![horizontal](https://user-images.githubusercontent.com/3615387/196596749-ef5bc3c1-26b7-4fd9-be3f-c3b5d5630f0b.jpg)

I would like it display as vertical as following. Can you double check the code?
![vertical](https://user-images.githubusercontent.com/3615387/196596781-f9b846d5-4d7e-426d-bbf5-d9a6ce651d74.jpg)

**Reply by YoshifumiFuyuno** [reply]
Are you sure you are setting the "Number of icons per row in grid" to 0?
Must be greater than or equal to 1 when using this CSS.
If you want to display them in a single row, you can use a large number.

**Reply by darkj2k** [reply]
My settings of ""Number of icons per row in grid" is 0 because I want the main toolbar in a single row (horizontal). And folder display as single column like context menu (vertical).

**Reply by YoshifumiFuyuno** [reply]
As I mentioned earlier, you can use large number.

**Reply by darkj2k** [reply]
I tried and tried but still can not figure out. :-( If set the "Number of icons per row in grid" to 1 then the top toolbar and folder will be vertical as single column. But I don't want the top toolbar as vertical. Just need vertical for folder. Can you export the options to me with single row top toolbar and vertical folder? Therefore I can figure out what's went wrong or something misunderstanding. Thanks in advance.

**Reply by YoshifumiFuyuno** [reply]
Because of the machine translation, it doesn't seem to be conveyed correctly.:cry:
If you set it to, say, 99 instead of 1, it should have the same effect as 0.
I hope that it will be conveyed well this time.:wink:

**Reply by YoshifumiFuyuno** [reply]
I made a video because I had a little time.
If I understand correctly, this is what you want to do.
Is this wrong?
Microsoft Edge v106 + new profile
<video controls src="https://user-images.githubusercontent.com/9511854/197334160-2e2c4b77-4773-4fa5-8466-8488fe0883d4.webm" muted="false"></video>

**Reply by darkj2k** [reply]
Thanks for your time and video. Yes, that's what I want. I also figure out what went wrong. I have an extra code for column width:
```
.spc-eFolder.s-iconAndName {
  --col: 1;
  --name-width: auto;
 }
```
With `--name-width: auto;` the folder displayed as horizontal not vertical. Removing `--name-width: auto;` make folder display as vertical perfectly. Anyway to making the folder displayed vertical but also with auto width function?

Also, as you see the separator in folder looks not normal. Anyway to fix it?
![separator](https://user-images.githubusercontent.com/3615387/197385329-7f15bd0c-e1f8-4995-ae32-a379203b69d3.jpg)

**Reply by YoshifumiFuyuno** [reply]
Like this?
```CSS
.spc-engines .name {
  --name-width: auto;
}
```
However, there may possibly be some problem because it is not designed to use auto.

```CSS
.separator {
  border-top: solid gray 2px;
  width: 100%;
}
.separator > .hoverSet {
  display: none;
}
```
I think the color and width need to be adjusted.

**Reply by darkj2k** [reply]
With this code:
```
.separator {
  border-top: solid gray 2px;
  width: 100%;
}
.separator > .hoverSet {
  display: none;
}
```
It looks like this:
![separator](https://user-images.githubusercontent.com/3615387/197447191-fb4d2fbe-b57a-4ed2-b8e5-ef240da62c6a.jpg)

Separator in folder is what I want but in top toolbar is not ok.

**Reply by YoshifumiFuyuno** [reply]
Ah, I forgot about the top level.
On second thought, if you use `--name-width`, you might be better off using `flex-direction`.
`--col: 1;` must be removed.
```CSS
.spc-eFolder {
  flex-direction: column;
  --name-width: auto;
}
```
* separator
```CSS
.spc-eFolder .separator {
  border-top: solid gray 2px;
  margin-top: 4px;
  width: 100%;
}
.spc-eFolder .separator > .hoverSet {
  display: none;
}
```

**Reply by darkj2k** [reply]
Great job. :-) You just make it better and better. My final question to make it perfect. 

Since there's no hotkey to show/hide the "search box and engine filter box" only option to enable/disable it. Possible to make a button to show/hide the "search box and engine filter box"?

**Reply by YoshifumiFuyuno** [reply]
I was able to do it at hand. However, it is difficult to create an icon for the button, so it is toggled by right-clicking in the margin.
I assume you want it to be hidden by default and only show when you need it, but I have a few questions.
1. does the show/hide toggle affect all tabs or just that tab?
2. is the show/hide toggle only at that time? In other words, when the popup is displayed again on the same page, is it hidden?

**Reply by darkj2k** [reply]
You can read my mind. :-P Yes, I want it hidden as default  and show it when I need it. Maybe you can add it as a toggle button in the future release if possible. For your question:

1. Just for that tab.
2. Yes, only at that time. If popup is displayed again on the same page it will keeping hidden.

Also, by your guide I know how to change the background color of folder. For the top toolbar, there's an color option for it. If I want to change the top popup background color by CSS code. How can I do it to override the color in option?

**Reply by YoshifumiFuyuno** [reply]
Which do you think is better?
* Test1
Toggle by right-clicking in the margin.
![test1](https://user-images.githubusercontent.com/9511854/197768194-11d66f42-979f-4e16-a268-4ce1649b48bc.gif)

* Test2
Click the bottom to toggle.
![test](https://user-images.githubusercontent.com/9511854/197754151-288eeb12-667e-4728-a4dc-42f30d0ba36d.gif)

Top-level background color. Like this.
```CSS
.spc-content {
  background: red;
}
```

**Reply by darkj2k** [reply]
I like method 1 by right-clicking in the margin. 👍

**Reply by YoshifumiFuyuno** [reply]
Thanks, got it. I will proceed with test1.:smiley:
If there are no problems, I will perhaps submit it next Sunday.

**Reply by darkj2k** [reply]
Good to know. Can't wait for it.
